openwisp_monitoring/device/utils.py

```diff
@@ -1,9 +1,24 @@
+from django.conf import settings
+from swapper import load_model
+
 from ..db import timeseries_db
 from . import settings as app_settings
+from .signals import device_status_unknown
 
 SHORT_RP = 'short'
 DEFAULT_RP = 'autogen'
 
+Device = load_model('config', 'Device')
+DeviceMonitoring = load_model('device_monitoring', 'DeviceMonitoring')
+
+
+def handle_critical_check_change(check):
+    critical_metrics = settings.OPENWISP_MONITORING_CRITICAL_DEVICE_METRICS
+    if check.object_id and check.metric.name in critical_metrics:
+        device = Device.objects.get(pk=check.object_id)
+        device.monitoring.update_status('unknown')
+        device_status_unknown.send(sender=DeviceMonitoring, instance=device.monitoring)
```

Comment:
we can eliminate this signal because the `update_status` method will take care of sending the `health_status_changed` signal which can be used to know when the health status has moved to unknown.